listeners.py

```diff
@@ -50,7 +57,48 @@ def process_result(self, obj):
                                       obj.result_checksum,
                                       obj.show_reproduce_content))
 
+    def get_failed_task_name(self, task):
+        return task[0][0] + ((':' + task[0][1]) if task[0][1] else '')
+
+    def get_long_mark(self, task):
+        return '(long)' if task in self.long_tasks else ''
+
     def print_statistics(self):
+        field_size = 60
+        top_rss = 10
+        results_sorted = []
+
+        # prepare standalone subpath '<vardir>/statistics' for statistics files
+        stats_dir = os.path.join(Options().args.vardir, 'statistics')
+        safe_makedirs(stats_dir)
+
+        # RSS
```

Comment:
I think we should skip this output (to the terminal and to the file both) when `/proc` is not available at all (on Mac OS). And I would not add the sampler wather at all in the case.

Reply:
Right, done.